src/Type/EntityQuery/EntityQueryType.php

```diff
@@ -22,4 +22,9 @@ public function withAccessCheck(): self
 
         return $type;
     }
+
+    protected function describeAdditionalCacheKey(): string
+    {
+        return $this->hasAccessCheck ? 'with-access-check' : '';
+    }
```

Comment:
Should we make sure that we has `without-access-check` as the other option for verbosity?